zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -332,7 +315,13 @@ public void testInitiateConnection(long sid) throws Exception {
         LOG.debug("Opening channel to server " + sid);
         Socket sock = new Socket();
         setSockOpts(sock);
-        sock.connect(self.getVotingView().get(sid).electionAddr, cnxTO);
+        InetSocketAddress address;
+        try {
```

Comment:
This logic is quite common across the board. It would be better located in a parent or helper class.
For example in the `MultipleAddresses` class something like `getReachableOrOne`.